packages/client/src/links/wsLink/wsClient/wsClient.ts

```diff
@@ -330,7 +330,10 @@ export class WsClient {
       handleCloseOrError(event);
       this.callbacks.onClose?.(event);
 
-      if (!this.lazyMode) {
+      if (
+        !this.lazyMode ||
+        (this.lazyMode && this.requestManager.hasPendingRequests())
+      ) {
         this.reconnect(
```

Comment:
Isn't this equivalent to 

```suggestion
      if (
        !this.lazyMode || this.requestManager.hasPendingRequests())
      ) {
        this.reconnect(
``` ?

Reply:
Yes, it is. My bad! 

Updated the PR to reflect this